src/main/java/org/jenkinsci/plugins/workflow/support/concurrent/ChainingListenableFuture.java

```diff
@@ -30,6 +29,7 @@
  * in a {@code UndeclaredThrowableException} so that this class can get
  * access to the cause.
  */
+@Deprecated
```

Comment:
This class is only used from `workflow-support`. Deprecating it so that `workflow-support` can prefer the upstream version once core is on a recent version of Guava and `workflow-support`'s baseline has been updated.